doc/deps.md

```diff
@@ -91,6 +93,19 @@ by other processes which could, in extreme cases, make the system very hard or
 impossible to recover. **Please do NOT add a `sudo` in front of any `make` or
 other commands below, unless *explicitly* specified!**
 
+### Odb
+Odb is an Object relational Mapping tool, that is required by CodeCompass.
+We are using odb 2.5 which is not part of the official ubuntu repositories, so
+it needs to be built manually. For this, we provide a convenience script, under
+the scripts directory.
+
+```bash
+cd CodeCompass/scripts
+chmod u+x install_odb_2.5.sh
+./install_odb_2.5.sh
+```
```

Comment:
I don't think that the install script should be "hidden" in the `install_odb_2.5.sh` file, as neither the Thrift, Google Test or previously LLVM installation was approached in such a way. (Long ago we had an *install dependencies* script file, it similar to that.)
Or to be more precise, it's confusing that there is an install script for ODB, but not for other dependencies (Thrift, Google Test). It should be unified.

The usage of the script is also undocumented, like configuring the install location.

Reply:
I'm working on the CI script but since I don't really know the ins and outs of the topic it may take some time.

And I will expand the documentation of the install_odb script, then too.
The reason why I wrote an install script rather than, a long and detailed description, on how to build odb 2.5 is that it has several steps, that can take a long time on a low end machine, and this would make then build process rather frustrating. And the odb build uses its own build system too `build2`, unlike thrift and googletest. Those can be installed via cmake, in a more conventional way.

Reply:
Agree, magic scripts is somewhat of a bad idea. It's horrible already that you cannot trust the upstream packages to not contain conflicts on requirements...